internal/buildinfo/buildinfo.go

```diff
@@ -17,9 +17,30 @@
 package buildinfo
 
 var (
-	// BuildID is the unique build identifier.
-	BuildID string = "unknown"
+	// id is the unique build identifier.
+	id string = "unknown"
```

Comment:
I'm 85% certain that variables that are set from LDFLAGS have to be public. I think you can leave this as-is (and revert the build.yaml changes) and still have the same desired effect.